### PR TITLE
New report --how html available

### LINKED ISSUE #441: [RFE] report --how html
Use case: tmt run for a lot of cases for later review.

It is easier to see fail/pass in browser and click for logs then to process text from terminal
Inspiration: restraint -> creates index.html with summary + links to logs

```tmt run report --how html --open``` could open the index page in default browser
Otherwise just a location of index will be printed.

The index.html (+css/js) could be placed to the workdir of 'report' step - No need to copy any of files as they are already there.

### PR DESCRIPTION
**Pull request by lukaszachy**
Resolves #441 

Very, very simple html site.
Ideas about usable css sheets and so on? It has been ages since I've played with HTML.

Maybe we can bundle (or process) beakerlib's output.log

### CODE REVIEW

File: tmt/steps/report/html.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            return
+
+        target = os.path.join(self.workdir, file_name)
+        self.info("Wrote report to", target)
+        if not self.opt('open'):
+            return
+
+        # opening target using xdg-open, fallback to $BROWSER
+        try:
+            process = subprocess.Popen(["xdg-open", target])
```
Comment (lukaszachy) [inline tmt/steps/report/html.py:102]: Unused variable, no need to assign it
```suggestion
            subprocess.Popen(["xdg-open", target])
```

**Review by lukaszachy — COMMENTED** [review]
@psss Feel free to resolve issues. 
I forget on packaging - probably new subpackage is necessary due jinja2 dependency.

**Review by psss — APPROVED** [review]
Thanks for implementing this, @lukaszachy.
Looks fine to me. Did a bunch of adjustments.
Please, have a look and check if you like the changes.

**Comment by psss** [comment]
Just one note: Using `Common.run()` provides some under-the-hood debugging for free. It also catches possible unwanted output like `Opening in existing browser session.` in this case. That's why I think it's better instead of calling `subprocess.Popen` directly.

**Comment by psss** [comment]
And yes, we should/could have a separate package for the `html` plugin but let's address that in a separate pull request.

**Comment by lukaszachy** [comment]
> Just one note: Using `Common.run()` provides some under-the-hood debugging for free. It also catches possible unwanted output like `Opening in existing browser session.` in this case. That's why I think it's better instead of calling `subprocess.Popen` directly.

I had troubles with no-exiting xdg-open on my system - it waited until I close the browser which is something I don't want.
It should be 'execute and forget':)

**Comment by psss** [comment]
Rebased on the latest master, going to merge. We'll address the blocking issue in #452.

**Review by psss — APPROVED** [review]

**Pull request merged**